### ISSUE TITLE
Errors in evaluation 

### ISSUE THREAD

**Issue by deepmeng** [matssteinweg/PIXOR#5]
When I loaded the well-trained model for evaluation (execute the file 'evaluate_model.py'), there was an obvious problem with the mAP, which is 'mAP>1'. This issue can be replicated when parameters in the function 'process_predictions()' are set to 'confidence=0.1, nms_threshold =0.51'. Some other parameters also meet the same error, such as 'conf = 0.1, nms=0.61', 'conf = 0.15, nms=0.66', 'conf = 0.5, nms=0.76', 'conf = 0.3, nms=0.61', etc. I suspect there may be some errors in the way of computing mAP. But I don't have an answer right now.

**Comment by deepmeng** [comment]
When I loaded the well-trained model for evaluation (execute the file 'evaluate_model.py'), there was an obvious problem with the mAP, which is 'mAP>1'. This issue can be replicated when parameters in the function 'process_predictions()' are set to 'confidence=0.1, nms_threshold =0.51'. Some other parameters also meet the same error, such as 'conf = 0.1, nms=0.61', 'conf = 0.15, nms=0.66', 'conf = 0.5, nms=0.76', 'conf = 0.3, nms=0.61', etc. I suspect there may be some errors in the way of computing mAP. But I don't have an answer right now.

**Comment by matssteinweg** [comment]
Thanks for pointing this out. I was not aware of any problems with the evaluation. Unfortunately, I don't have time to look into it at the moment. If you find the error, I'd appreciate it if you let me know.

**Comment by deepmeng** [comment]
I debugged the source code and I analyzed the possible reasons which were thought to cause the pointed error. Now, I have ruled out your code and the calculation methods. I found that when the NMS threshold is too large, it will keep more prediction results than a small NMS threshold. Thus, more prediction results will be kept after compared with the ground truth instances. Finally, the values of recall and mAP will be inaccurate.
